framework/src/modules/chain/logic/bft.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const assert = require('assert');
+
+const __private = {
+	items: new WeakMap(),
+};
+
+class HeadersList {
+	constructor({ size }) {
```

Comment:
We agreed that BFT can rollback an indefinite number of blocks, and so the headers list should have not have an initialization size of the list. Rather, it should maintain a list which is dynamic.

Reply:
List have fixed size, but you can still add headers from previous height. So you are on 500 blocks, you removed 450 and added 200 more from past. So the list size will be 250 now. But the actual size of the list will strict not load up million block headers in memory.

Reply:
I don't understand your comment. Can you explain it in a different way? Also, what will be the appropriate initialization value for the HeadersList in BFT according to you?

Reply:
We initialize the list with 500 blocks, in certain cases if we need a larger list we can extend it by just setting the new size `list.size = 10000`.